lib/OpenQA/WebAPI/Controller/Admin/JobTemplate.pm

```diff
@@ -21,12 +21,22 @@ sub index {
     my ($self) = @_;
 
     my $schema = $self->schema;
-    $self->stash('group', $schema->resultset("JobGroups")->find($self->param('groupid')));
+    my $group  = $schema->resultset('JobGroups')->find($self->param('groupid'));
+    return $self->reply->not_found unless $group;
+
+    my $yaml = $group->template;
+    my $force_yaml_editor
+      = defined $yaml || $schema->resultset('JobTemplates')->search({group_id => $group->id}, {rows => 1})->count == 0;
```

Comment:
This probably means that the YAML editor will also show up after removing everything in it?

Maybe as an alternative we could change the default for new groups when they're created. It might also not be a concern, but I figured I'd bring it up in any case.

Reply:
Yes, after emptying a group (and reloading the page) you will have to continue with YAML. I guess that's ok, though. People are usually emptying groups when they want to delete them anyways.